### ISSUE TITLE
pftool include list doesn't work

### ISSUE THREAD

**Issue by gsparrow** [pftool/pftool#50]
When using PFTool and include lists together, things do not work. When listing files with an include list, when the -vv option is also set, pftool does a DATASTAT of the files, but returns the size in number of bytes of the input file, rather than the combined size of the files in the input file. In Jenkins this is test "PFTool_list_three_MarFS_unifiles_from_POSIX_input_file" and also test "PFTool_list_four_Posix_unifiles_from_POSIX_input_file". include lists on copy with cp and compare with include lists also do not work, and result in a non-zero exit status of PFTool. These are in Jenkins as "PFTool_copy_a_unifile_from_MarFS_to_POSIX_using_an_input_list_prove_using_PFTool", "PFTool_compare_two_different_MarFS_unifiles_using_an_input_list" and "PFTool_compare_the_same_MarFS_unifile_using_an_input_list". Below is a simple reproducer of the bug with ls.

```
export P=/marfs/testing/test.txt
export P2=/marfs/testing/test.2.txt
export P3=/marfs/testing/test.3.txt
export INPUT_FILE=./input_file.txt

echo "Hello World" > $P
echo "Hello Worl" > $P2
echo "Hello Wor" > $P3

echo $P > $INPUT_FILE
echo $P2 >> $INPUT_FILE
echo $P3 >> $INPUT_FILE

sleep 10

mpirun -np $NP $PFTOOL -w 1 -i $INPUT_FILE
```

**Comment by cadejager** [comment]
I ran a test on a directory with 1M files. If I use a list of all the files in the directory it finds the size of each file to just be the size of the filelist (151 bytes) but if I list the directory recursively then I get the correct size. It is also using the permissions for the filelist and not the actual files.

```
dejager@hbc022:~/src/pftool/installed/bin$ mpirun -np 4 ./pftool -w 1 -r -p ~/testfiles/1Mzeros/ -vv
INFO  HEADER   ========================  TestJob  ============================
INFO  HEADER   Starting Path: /home/dejager/testfiles/1Mzeros/
INFO  HEADER   Source-type: POSIX_Path
INFO  HEADER   Dest-type:   POSIX_Path
RANK   3: INFO  DATASTAT - drwxr-xr-x      8  26158  26158                  4096 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.10
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.11
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.8
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.4
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.14
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.5
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.1
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.13
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.7
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.2
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.16
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.3
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.12
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.9
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.15
RANK   3: INFO  DATASTAT - -rw-r--r--      0  26158  26158               1048576 Thu Aug 18 2016 08:44:08 /home/dejager/testfiles/1Mzeros/1Mzero.6
INFO  FOOTER   ========================   NONFATAL ERRORS = 0   ================================
INFO  FOOTER   =================================================================================
INFO  FOOTER   Total Files/Links Examined: 16
INFO  FOOTER   Total Bytes Examined: 16777216
INFO  FOOTER   Total Dirs Examined: 1
INFO  FOOTER   Elapsed Time: 0 seconds
dejager@hbc022:~/src/pftool/installed/bin$ mpirun -np 4 ./pftool -w 1  -i ~/filelist.txt  -vv
INFO  HEADER   ========================  TestJob  ============================
INFO  HEADER   Starting Path:
INFO  HEADER   Source-type: POSIX_Path
INFO  HEADER   Dest-type:   POSIX_Path
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.1
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.10
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.11
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.12
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.13
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.14
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.15
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.16
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.2
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.3
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.4
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.5
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.6
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.7
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.8
RANK   3: INFO  DATASTAT - -rwx------      8  26158  26158                   151 Thu Nov 10 2016 09:27:18 1Mzero.9
INFO  FOOTER   ========================   NONFATAL ERRORS = 0   ================================
INFO  FOOTER   =================================================================================
INFO  FOOTER   Total Files/Links Examined: 16
INFO  FOOTER   Total Bytes Examined: 2416
INFO  FOOTER   Total Dirs Examined: 0
INFO  FOOTER   Elapsed Time: 0 seconds
dejager@hbc022:~/src/pftool/installed/bin$
```

**Comment by cadejager** [comment]
I believe that this was completely fixed as of commit c1fc926300333ec51f0d587fd38c18238d2ce59a.
